### ISSUE TITLE
Generated barcode not recognized

### ISSUE THREAD

**Issue by mbaros100** [zxing-cpp/zxing-cpp#495]
I have generated the below Code39 barcode using the zxing-cpp library and maybe did a bit of post processing on the dimensions.

All online barcode readers including the zxing (https://zxing.org/w/decode.jspx) are able to decode the barcode but not our core library.
![aa](https://user-images.githubusercontent.com/8537079/214157556-59c1de5e-00ad-4d92-a0e6-a46d02e95805.png)

**Comment by axxel** [comment]
> did a bit of post processing on the image dimensions

That is exactly the problem. You did not follow the advice that I gave you [here](https://github.com/zxing-cpp/zxing-cpp/discussions/482#discussioncomment-4650941). If you render the symbol to the minimal size and then sale it up by a non-integral scale factor that is below 4 (I wrote '10' in the mentioned discussion but it probably does not need to be that high. Where exactly the limit is, depends on the used symbology and the tolerance of the decoder). Here the scale factor seems to be something between 1 and 2 and that is a really idea.

That said, the QRCode sample will likely work in an upcoming release but I strongly recommend to change your scaling approach none-the-less.

@gitlost How often do people complain about non-working images from zint that are caused by an inappropriate scaling?

**Comment by gitlost** [comment]
Unfortunately I get very little feedback (could end the sentence there) about scaling and non-working images - can't find a relevant ticket in the last couple of years.

**Comment by axxel** [comment]
> So we definitely have a room for being more robust and handling these cases!

I agree. I'm surprised that zxing.org is capable of detecting these but can not argue with that fact. Interestingly, if I try to scan those off of my screen with our own android app, it fails if I set it to use the zxing Java module, which is somewhat surprising considering the zxing.org results. I also did not have a look at where exactly the Code39 symbol fails. For the QRCode, there is light on the horizon as I said.

> Our software is trying to have a maximal unified size for all the 1d barcodes in order to preserve consistency for the users which we chose to be 210x70

While that may be convenient for the user, I don't think you are making their or your life easier in the long-run. That will render your output 100% broken if your input for a variable sized code (like Code 39) is growing above a certain threshold, regardless of which scanner you use. If you want to have a fixed size and your barcodes should have the same sized quiet zone everywhere, the width of the image needs to be a lot bigger _and_ you still have to limit the input length that you want to encode.

This is precisely the reason why `ZXing::Inflate` behaves as it does and I recommend you use that approach, even if it "looks ugly" but therefore it has the highest chance of actually working, regardless of the scanner that is involved.

**Comment by mbaros100** [comment]
@axxel what about this barcode? I generated using zxing-cpp and I haven't touched the size (data="1234" type=ITF). 
The reader is not able to detect any barcodes. For the reference java version is also not able to detect anything but the website https://online-barcode-reader.inliteresearch.com/ and some others are successfully decoding and returning a type "Interleaved2of5"
 
![a](https://user-images.githubusercontent.com/8537079/216201380-1abb7ac5-23be-4305-bd12-923c53a70547.png)

**Comment by axxel** [comment]
The reason why it is not detected is [the `minCharCount = 6` line](https://github.com/zxing-cpp/zxing-cpp/blob/5142600c5d6f47b0705697a84f8c4d508c3a22e8/core/src/oned/ODITFReader.cpp#L24). This number is technically arbitrary but it is there to prevent lots of false positives inside other symbologies. If I set this to 4, it detects your sample but breaks 6 test from the blackbox tests because phantom ITF sybols are found.

**Comment by mbaros100** [comment]
Understand. It is a little bit weird situation that the library is not able to recognize an own generated barcode. 

I think i will limit it to 6+ characters in our software to prevent user confusion and issue a special message when input is less than that.

thanks

**Comment by axxel** [comment]
> It is a little bit weird situation that the library is not able to recognize an own generated barcode.

Absolutely. This is due to the history of this project. I'm hesitant to put any work in the generator though, since I intend to replace all that with the much more capable libzint based backend.
